### DISCUSSION TITLE
Change frontend port when using OIDC SSO

### DISCUSSION THREAD

**Discussion by rjvrvjr** [Velocidex/velociraptor#3539] [Q&A]
I'm running a debian server with SSO enabled with OIDC where i want to run GUI on port 5000 and frontend on 8000 (basically anything other than 80 or 443). I have changed the bind_port fro both in my server configuration file but for some reason only GUI is getting binded to it's custom port and frontend is still running on 443. Is there any thing else I have to change to get that implemented or it'll only run on 443.

**Config:**

Frontend:
  hostname: <IP Address>
  bind_address: 0.0.0.0
  bind_port: 8000
  certificate: |
  private_key: |
  dyn_dns: {}
  default_client_monitoring_artifacts:
  run_as_user:
  GRPC_pool_max_size:
  GRPC_pool_max_wait:
  resources:
    connections_per_second:
    notifications_per_second:
    max_upload_size:
    expected_clients: 

**Logs:**

[INFO] 2024-06-05T16:50:06Z Throttling connections to 100 QPS
[INFO] 2024-06-05T16:50:06Z Starting gRPC API server on 127.0.0.1:8001
[INFO] 2024-06-05T16:50:06Z Launched Prometheus monitoring server on 127.0.0.1:8003
[INFO] 2024-06-05T16:50:06Z Autocert is enabled but GUI port is not 443, starting Frontend with autocert and GUI with self signed.
[INFO] 2024-06-05T16:50:06Z GUI will use the OIDC authenticator
[INFO] 2024-06-05T16:50:06Z GUI is ready to handle TLS requests on https://<IP Address>:5000/
[INFO] 2024-06-05T16:50:06Z Frontend is ready to handle client requests at https://<IP Address>/
[ERROR] 2024-06-05T16:50:06Z Frontend server: Can not listen on :https: listen tcp :443: bind: address already in use
[INFO] 2024-06-05T16:50:07Z FrontendService: Watching for events from Server.Internal.FrontendMetrics
[INFO] 2024-06-05T16:50:07Z Compiled all artifacts.

**Comment by scudette** [comment]
Your log suggests you are using autocert to mint let's encrypt certificates. Those can only work on port 443 so the server ignores your bind port directive and overrides it with 443.

If you want to bind to another port you will need to use self signed certs or get certificates from another provider. 

https://docs.velociraptor.app/docs/deployment/security/#deployment-signed-by-lets-encrypt
